### PR TITLE
Restore docs: updated restore tables documentation

### LINKED ISSUE #3488: Document MV and SI restore procedure
A follow-up docs issue related to #3466.

### LINKED ISSUE #3489: Document post-restore repair procedure
A follow-up docs issue related to #3414.

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
This PR is a follow-up to restore changes introduced in SM 3.2.

Related #3479
Fixes #3489
Fixes #3488

### CODE REVIEW

File: docs/source/restore/restore-tables.rst [hunk 1]
```diff
@@ -4,63 +4,36 @@ Restore tables
 
 .. note:: Currently, Scylla Manager does not support restoring content of `CDC log tables <https://docs.scylladb.com/stable/using-scylla/cdc/cdc-log-table.html>`_.
 
+.. note::
+   The ``--keyspace`` flag defines base tables which will be restored.
```
Comment (annastuchlik) [inline docs/source/restore/restore-tables.rst:8]:
```suggestion
   The ``--keyspace`` flag defines base tables that will be restored.
```

Comment (annastuchlik) [reply docs/source/restore/restore-tables.rst:8]: The content of this note is clear, but the usage is not so much. 
Is the user expected to run:
`sctool restore --cluster <id|name> --location [<dc>:]<provider>:<bucket> --snapshot-tag <tag> [flags] ---restore-tables --keyspace`
?
Is adding `--keyspace` required?

I THINK this section should be modified like this:

- (If --keyspace is optional)

_To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag._

_You **can** add the ``--keyspace`` flag to define the base tables that will be restored.  All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure the correct data state._

- (If --keyspace is mandatory)

_To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag._

_**Add** the ``--keyspace`` flag to define the base tables that will be restored.  All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure the correct data state._

File: docs/source/restore/restore-tables.rst [hunk 2]
```diff
@@ -4,63 +4,36 @@ Restore tables
...
+
+   All Materialized Views and Secondary Indexes of those tables will be automatically recreated in order to ensure correct data state.
+
 To restore the content of the tables (rows), use the :ref:`sctool restore <sctool-restore>` command with the ``--restore-tables`` flag.
 
 Prerequisites
 =============
 
-* Scylla Manager requires CQL credentials with `permission to alter <https://docs.scylladb.com/stable/operating-scylla/security/authorization.html#permissions>`_ restored tables.
+* Scylla Manager requires CQL credentials with:
```
Comment (annastuchlik) [inline docs/source/restore/restore-tables.rst:17]:
```suggestion
* Scylla Manager requires CQL credentials with:

```

Comment (annastuchlik) [reply docs/source/restore/restore-tables.rst:17]: You must add an empty line for this section to render correctly.

File: docs/source/restore/restore-tables.rst [hunk 3]
```diff
@@ -4,63 +4,36 @@ Restore tables
...
+    * `permission to alter <https://opensource.docs.scylladb.com/stable/operating-scylla/security/authorization.html#permissions>`_ restored tables.
+    * `permission to drop and create <https://opensource.docs.scylladb.com/stable/operating-scylla/security/authorization.html#permissions>`_ Materialized Views and Secondary Indexes of restored tables.
 
-* Restoring the content of the tables assumes that the correct schema is already present in the destination cluster.
+* Restoring the content of the tables assumes that the correct schema (identical as in the backup) is already present in the destination cluster.
```
Comment (annastuchlik) [inline docs/source/restore/restore-tables.rst:21]:
```suggestion
* Restoring the content of the tables assumes that the correct schema (identical as in the backup) is already present in the destination cluster.

```

Comment (annastuchlik) [reply docs/source/restore/restore-tables.rst:21]: You must add an empty line for this section to render correctly.

File: docs/source/restore/restore-tables.rst [hunk 4]
```diff
@@ -4,63 +4,36 @@ Restore tables
...
    Scylla Manager does NOT validate that, so it's user responsibility to ensure it. The only form of validation
    that Scylla Manager performs is checking whether restored tables are present in destination cluster,
    but it does not validate their columns types nor other properties. In case destination cluster is missing correct schema,
    it should be :doc:`restored <restore-schema>` first.
 
+   It's important to note, that this requirement also involves keyspace schema.
```
Comment (annastuchlik) [inline docs/source/restore/restore-tables.rst:27]:
```suggestion
   Note that this requirement also applies to the keyspace schema.
```

File: docs/source/restore/restore-tables.rst [hunk 5]
```diff
@@ -4,63 +4,36 @@ Restore tables
...
+   Keyspace (where restored tables live) should have correct schema, even in terms of replication strategy.
```
Comment (annastuchlik) [inline docs/source/restore/restore-tables.rst:28]:
```suggestion
   Keyspace (where restored tables live) should have the correct schema, including the replication strategy.
```

**Review by annastuchlik — COMMENTED** [review]
I've left some suggestions so that the language is correct and the page renders correctly.
In addition, I'd like to clarify the first part regarding the usage of --keyspace - please see the comments.

**Comment by Michal-Leszczynski** [comment]
@annastuchlik I fixed things mentioned in the comments, please take a look.

**Review by annastuchlik — APPROVED** [review]
LGTM

**Review by karol-kokoszka — APPROVED** [review]

**Pull request merged**